logic/src/commonMain/kotlin/com/wire/kalium/logic/network/SessionManagerImpl.kt

```diff
@@ -60,4 +66,15 @@ class SessionManagerImpl(
     override suspend fun onClientRemoved() {
         sessionRepository.logout(userId, LogoutReason.REMOVED_CLIENT)
     }
+
+    override fun proxyCredentials(): ProxyCredentialsDTO? =
+        wrapStorageNullableRequest { proxyCredentialsStorage.getProxyCredentials() }.nullableFold({
```

Comment:
I can't get this part, we have the nullableFold then we check the nullability manually in next lines!

Reply:
isn't the nullable to return null if there is no value !, so here the storage can be null before we store any credentials or in the case where we don't have proxy at all , so we make it nullable and check if the value not null so we map it else return null